src/lib/PostgresMetaTriggers.ts

```diff
@@ -0,0 +1,292 @@
+    const { name: currentName, schema, table } = triggerRecords![0]
+    const qualifiedTableName = `${ident(schema)}.${ident(table)}`
+    const updateNameSql = newName
+      ? `ALTER TRIGGER ${ident(currentName)} ON ${qualifiedTableName} RENAME TO ${ident(newName)};`
+      : ''
+    const updateExtDepSql =
+      extension && ['true', 'false'].includes(is_extension_dependent || '')
+        ? `ALTER TRIGGER ${ident(currentName)} ON ${qualifiedTableName} ${
+            is_extension_dependent === 'false' ? 'NO' : ''
+          } DEPENDS ON EXTENSION ${ident(extension)};`
+        : ''
+
+    if (is_enabled === 'false') {
+      enableDisableSql = `ALTER TABLE ${qualifiedTableName} DISABLE TRIGGER ${ident(currentName)};`
+    }
+
+    if (is_enabled === 'true') {
+      const enableMode =
+        (enable_mode && ['replica', 'always'].includes(enable_mode) && ident(enable_mode)) || ''
+      enableDisableSql = `ALTER TABLE ${qualifiedTableName} ENABLE ${enableMode} TRIGGER ${ident(
+        currentName
+      )};`
+    }
+
+    // updateNameSql must be last
+    const sql = `BEGIN; ${enableDisableSql} ${updateExtDepSql} ${updateNameSql} COMMIT;`
+
+    {
+      const { error } = await this.query(sql)
+
+      if (error) {
+        return { data: null, error }
+      }
+    }
+
+    return await this.retrieve({ id })
+  }
+
+  async remove(id: number, { cascade = false }): Promise<PostgresMetaResult<PostgresTrigger[]>> {
+    const { data: triggerRecords, error } = await this.fetchTriggersById(id)
+
+    if (error) {
+      return { data: null, error }
+    }
+
+    const { name, schema, table } = triggerRecords![0]
+    const qualifiedTableName = `${ident(schema)}.${ident(table)}`
+    const sql = `DROP TRIGGER ${ident(name)} ON ${qualifiedTableName} ${cascade ? 'CASCADE' : ''};`
+
+    {
+      const { error } = await this.query(sql)
+
+      if (error) {
+        return { data: null, error }
+      }
+    }
+
+    return { data: triggerRecords!, error: null }
+  }
+
+  private async fetchTriggersById(id: number): Promise<PostgresMetaResult<PostgresTrigger[]>> {
+    const { data: triggers, error } = await this.retrieve({ id })
+    const triggerRecord = Array.isArray(triggers) && triggers[0]
+
+    if (error || !triggerRecord) {
+      return { data: null, error: error || { message: `Trigger with id ${id} does not exist` } }
+    }
+
+    return { data: triggers!, error: null }
+  }
+
+  private splitQualifiedName(schemas: string[], fullName: string): string[] {
+    const names = []
+    const sortedDescSchemas = schemas.sort((a, b) => b.length - a.length)
+
+    for (const schema of sortedDescSchemas) {
+      const startIdx = fullName.indexOf(schema)
+      const splitIdx = schema.length
+
+      if (startIdx === 0 && fullName[splitIdx] === '.') {
+        names.push(fullName.slice(0, splitIdx))
+        names.push(fullName.slice(splitIdx + 1))
+        break
+      }
+    }
+
+    return names[0] ? names : [fullName]
+  }
+}
```

Comment:
@soedirgo this is something we discussed a few days ago about escaping qualified table names. Even though this has to make an additional query for existing schemas, thought it covers the edge case when schema name itself has period(s). It doesn't cover every edge case of qualified table names but I think it's more than good enough for now. Let me know what you think!